portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -65,78 +65,80 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-   ArrayList<Message> tasks = new ArrayList<>();
+   ArrayList<Comment> tasks = new ArrayList<>();
 
     for (Entity entity : results.asIterable()) {
-    Message newMessage = new Message((Key) entity.getKey(),
-                                        (String) entity.getProperty("text"),
-                                        (long) entity.getProperty("timestamp"));
+
+      Key uniqueKey = (Key) entity.getKey();
+      String comment = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
+      String picture = (String) entity.getProperty("imageUrl");
+
+    Comment newMessage = new Comment(uniqueKey, comment, picture, timestamp);
         
-     // String picture = (String) entity.getProperty("imageUrl");
-     
       tasks.add(newMessage);
-      //tasks.add(pic);
     }
     
     Gson gson = new Gson();
     // Respond with the result.
-    response.setContentType("text/html;");
+     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(tasks));
     }
 
    @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
     
-    String text = request.getParameter("text-input");
+    String comment = request.getParameter("message");
+    String imageUrl = getUploadedFileUrl(request, "image");
```

Comment:
General question, not for this PR...
When you decide to implement comment deletion from datastore, how do you delete associated blobstore entry?

Reply:
I would add an event listener that would delete the imageurl from the specific comment

Reply:
Deleting imageurl doesn't delete blob contents. You will still have the image data stored in the blobstore. The thing is that image url only allows you to show the image on the client. To be able to delete this image later you need to store blobstore key somewhere.